Tests/xcodeprojMeasurementsTests/XcodeProjGenerator.swift

```diff
@@ -0,0 +1,144 @@
+    }
+    
+    @discardableResult
+    func generate(path: Path, sourceRoot: Path, setting: GenerateSetting = GenerateSetting()) throws -> XcodeProj {
+        let workspace = XCWorkspace()
+        let buildConfigurationList = XCConfigurationList()
+        let pbxproj = PBXProj(rootObject: nil, objectVersion: 0, archiveVersion: 0, classes: [:], objects: [])
+        let rootGroupName = UUID().uuidString
+        let rootGroupPath = sourceRoot + Path(rootGroupName)
+        try rootGroupPath.mkpath()
+        let mainGroup = PBXGroup(
+            children: [],
+            sourceTree: .group,
+            name: rootGroupName,
+            path: rootGroupName
+        )
+        pbxproj.add(object: mainGroup)
+        let project = PBXProject(
+            name: UUID().uuidString,
+            buildConfigurationList: buildConfigurationList,
+            compatibilityVersion: "Xcode 9.3",
+            mainGroup: mainGroup
+        )
+        pbxproj.add(object: project)
+        pbxproj.rootObject = project
+        let xcodeproj = XcodeProj(workspace: workspace, pbxproj: pbxproj, sharedData: nil)
+        xcodeproj.pbxproj.add(object: buildConfigurationList)
+        try fillProject(
+            pbxproj: xcodeproj.pbxproj,
+            rootProject: project,
+            sourceRoot: sourceRoot,
+            mainGroup: mainGroup,
+            setting: setting
+        )
+        try xcodeproj.write(path: path)
+        return xcodeproj
+    }
+    
+    private func fillProject(
+        pbxproj: PBXProj,
+        rootProject: PBXProject,
+        sourceRoot: Path,
+        mainGroup: PBXGroup,
+        setting: GenerateSetting) throws
+    {
+        try (0..<setting.targetCount).forEach { _ in
+            let target = generateTargets(pbxproj: pbxproj)
+            rootProject.targets.append(target)
+            var buildFiles = [PBXBuildFile]()
+            let currentPath = try mainGroup.fullPath(sourceRoot: sourceRoot)
+            try generateGroupsAndFiles(
+                for: mainGroup,
+                pbxproj: pbxproj,
+                sourceRoot: sourceRoot,
+                currentPath: currentPath!,
+                buildFiles: &buildFiles,
+                groupCount: setting.groupCountInLevel,
+                fileCount: setting.fileCountInLevel,
+                maxDepthLevel: setting.maxGroupDepth,
+                currentDepthLevel: 0
+            )
+            generateBuildPhase(
+                pbxproj: pbxproj,
+                target: target,
+                files: buildFiles
+            )
+        }
+    }
+    
+    private func generateTargets(pbxproj: PBXProj) -> PBXTarget {
+        let target = PBXNativeTarget(name: UUID().uuidString)
+        pbxproj.add(object: target)
+        return target
+    }
+    
+    @discardableResult
+    private func generateBuildPhase(pbxproj: PBXProj, target: PBXTarget, files: [PBXBuildFile]) -> PBXSourcesBuildPhase {
+        let buildPhase = PBXSourcesBuildPhase(
```

Comment:
Some of those attributes have default values so you can simplify this initialization a lot:

```swift
let buildPhase = PBXSourcesBuildPhase(files: files)
```